### ISSUE TITLE
"Name collision detected" error when using custom analyzer with stopwords

### ISSUE THREAD

**Issue by davechallis** [arangodb/arangodb#9652]
## My Environment

* __ArangoDB Version__:        3.5 RC7
* __Storage Engine__:             RocksDB
* __Deployment Mode__:       Single Server
* __Deployment Strategy__:   ArangoDB Starter in Docker

## Component, Query & Data

__Affected feature__:

Search views

## Steps to reproduce

1. create a custom analyzer with stopwords 
2. add that analyzer to a search view

__Problem__:

When creating a custom analyzer which contains stopwords, attempting to add it to a search view appears to give a `Name collision detected` error. Creating an identical analyzer without any stopwords works as expected.

When creating an analyzer in the CLI with:

```
127.0.0.1:8529@_system> a = require("@arangodb/analyzers");
{
  "ArangoAnalyzer" : [Function "ArangoAnalyzer(dat" ...],
  "analyzer" : [Function "(nam" ...],
  "remove" : [Function "(name, forc" ...],
  "save" : [Function "(name, type, properties, feature" ...],
  "toArray" : [Function "" ...]
}

127.0.0.1:8529@_system> a.save('custom_analyzer_a', 'text', {locale: 'en.UTF-8', case: 'lower', stopwords: ['the', 'of', 'inc', 'co', 'plc', 'ltd', 'ag'], accent: false, stemming: false}, ['position', 'norm','frequency']);
{
  "name" : "_system::custom_analyzer_a",
  "type" : "text",
  "properties" : {
    "locale" : "en.utf-8",
    "case" : "lower",
    "stopwords" : [
      "ag",
      "of",
      "plc",
      "the",
      "inc",
      "co",
      "ltd"
    ],
    "accent" : false,
    "stemming" : false
  },
  "features" : [
    "position",
    "norm",
    "frequency"
  ]
}
```

then adding it to a search view with the given `links` section:

```
 "links": {
    "assignee": {
      "analyzers": [
        "identity"
      ],
      "fields": {
        "name": {
          "analyzers": [
            "custom_analyzer_a"
          ]
        }
      },
      "includeAllFields": false,
      "storeValues": "none",
      "trackListPositions": false
    },
    "organisation": {
      "analyzers": [
        "identity"
      ],
      "fields": {
        "name": {
          "analyzers": [
            "custom_analyzer_a"
          ]
        }
      },
      "includeAllFields": false,
      "storeValues": "none",
      "trackListPositions": false
    }
  },
```

Gives the following error in the server's logs:

```
WARNING [ed6a3] {arangosearch} failure to get arangosearch analyzer name '_system::custom_analyzer_a': 10 Name collision detected while registering an arangosearch analyzer.
Current definition is:
{
  name:'_system::custom_analyzer_a'
  type: 'text'
  properties:'{
  "locale" : "en.utf-8",
  "case" : "lower",
  "stopwords" : [
    "ag",
    "of",
    "plc",
    "the",
    "inc",
    "co",
    "ltd"
  ],
  "accent" : false,
  "stemming" : false
}'
  features: [
    'position',
    'norm',
    'frequency'
  ]
}
Previous definition was:
{
  "features" : [
    "position",
    "norm",
    "frequency"
  ],
  "name" : "_system::custom_analyzer_a",
  "properties" : {
    "accent" : false,
    "case" : "lower",
    "locale" : "en.utf-8",
    "stemming" : false,
    "stopwords" : [
      "ag",
      "of",
      "plc",
      "the",
      "inc",
      "co",
      "ltd"
    ]
  },
  "type" : "text"
}
```

If I then remove the analyzer, and recreate it with `stopwords=[]`, then adding it to a search view works fine without any errors.

**Comment by Dronplane** [comment]
@davechallis   issue is currently fixed in devel by PR #9697

**Comment by Simran-B** [comment]
v3.5.1 was released, so closing.
